hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapreduce/lib/output/FileOutputCommitter.java

```diff
@@ -158,6 +158,11 @@ public FileOutputCommitter(Path outputPath,
         "output directory:" + skipCleanup + ", ignore cleanup failures: " +
         ignoreCleanupFailures);
 
+    if (algorithmVersion == 1 && skipCleanup) {
+        LOG.warn("Skip cleaning up when using FileOutputCommitter V1 can lead to unexpected behaviors. " +
+                "For example, committing several times may be allowed falsely.");
```

Comment:
"Skip cleaning up when using FileOutputCommitter V1 may corrupt the output".

there's another option here: we just ignore the setting on v1 jobs?
it's only there because directory deletion is so O(files) on GCS, and it targets v2 because that same file-by-file operation means that directory rename is never atomic; you may as well use the already unsafe v2 algorithm.

Reply:
ok, let's fail fast